File: C++/easy/1018.cpp
#include <iostream>
#include <vector>



class Solution {

public:
    std::vector<bool> prefixesDivBy5(std::vector<int>& nums) {
        std::vector<bool> v;
        long accum = 0;
        for (int i = 0; i < nums.size(); i++){
           accum = accum * 2 + nums[i];
           v.push_back(accum % 5 == 0);
        }
        return v;
    }
};

File: Python/Easy/1018.py
from typing import List 

class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        result = [0] * len(nums)
        accum = 0
        for index, bit in enumerate(nums):
            accum = accum * 2 + bit
            if accum % 5 == 0:
                result[index] = True
            else:
                result[index] = False
        return result